docs/marine-book/marine-runtime/api/marine-js-api.md

```diff
@@ -1,72 +1,126 @@
+        private logFunction: LogFunction,
+        private serviceConfig: MarineServiceConfig,
+        env?: Env,
+    ) 
 ```
+Constructs a `MarineService` object that is ready to be started. The 
 
-Locates WASM file in the specified npm package and loads it. This function can only be used in nodejs. Trying to call it inside browser will result throw an error.
-
-### loadWasmFromServer
+The service module bytes, names and all other possible configuration is passed as `MarineServiceConfig`. Acually used values are `module_bytes`, `import_names` and logger and WASI related ones. 
+```javascript
+export interface MarineServiceConfig {
+    /**
+     * Settings for a module with particular name (not HashMap because the order is matter).
+     */
+    modules_config: Array<ModuleDescriptor>;
+
+    /**
+     * Settings for a module that name's not been found in modules_config.
+     */
+    default_modules_config?: MarineModuleConfig;
+}
+
+export interface ModuleDescriptor {
+    wasm_bytes: Uint8Array;
+    import_name: string;
+    config: MarineModuleConfig;
+}
+
+export interface MarineModuleConfig {
+    /**
+     * Maximum memory size accessible by a module in Wasm pages (64 Kb).
+     */
+    mem_pages_count?: number;
+
+    /**
+     * Maximum memory size for heap of Wasm module in bytes, if it set, mem_pages_count ignored.
+     */
+    max_heap_size?: number;
+
+    /**
+     * Defines whether FaaS should provide a special host log_utf8_string function for this module.
+     */
+    logger_enabled: boolean;
+
+    /**
+     * Export from host functions that will be accessible on the Wasm side by provided name.
+     */
+    // host_imports: Map<string, HostImportDescriptor>;
+
+    /**
+     * A WASI config.
+     */
+    wasi: MarineWASIConfig;
+
+    /**
+     * Mask used to filter logs, for details see `log_utf8_string`
+     */
+    logging_mask: number;
+}
+
+export type Env = WASIEnv;
+
+export type Args = WASIArgs;
+
+export interface MarineWASIConfig {
+    /**
+     * A list of environment variables available for this module.
+     */
+    envs: Env;
+
+    /**
+     * A list of files available for this module.
+     * A loaded module could have access only to files from this list.
+     */
+    preopened_files: Set<string>;
+
+    /**
+     * Mapping from a usually short to full file name.
+     */
+    mapped_dirs: Map<String, string>;
+}
 
-```rust
-export const loadWasmFromServer = async (filePath: string): 
-    Promise<SharedArrayBuffer | Buffer>
 ```
-
-Loads WASM file from the service hosting the application. It accepts the file path on the server and returns a buffer compatible with `FluencePeer` API. This function can only be used in a browser. Trying to call it inside node.js will result throw an error.
-
-## Playing with FluenceAppService
-
-### initialization
-
+A marine service can produce logs during execution. All of its logs will be passed to the `logFunction` argument, which has the following type:
```

Comment:
```suggestion
A marine service can produce logs during execution. All of these logs will be passed into the `logFunction` function, which has the following type:
```

Reply:
Logs generated by a marine service are passed to the `logFunction`, which has the following type: